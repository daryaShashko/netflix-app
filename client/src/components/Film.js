import React from 'react';
import {FilmDescriptionItem} from './FilmDescriptionItem.js';
import {FilmPoster} from './FilmPoster.js';

export default class Film extends React.Component {
  render() {
    return (
      <div>
        <FilmPoster poster="http://netflixroulette.net/api/posters/60031236.jpg" show_title="Rocket boy"/>

        <header className="media-show__header">
          <FilmDescriptionItem className="media-show__title" text="Pulp Fiction"/>
          <FilmDescriptionItem className="media-show__rating" text="4.1"/>
          <FilmDescriptionItem className="media-show__category" text="Oscar-winning Movie"/>
          <FilmDescriptionItem className="media-show__release-year" text="1994"/>
          <FilmDescriptionItem className="media-show__runtime" text="154 min"/>
        </header>

        <div className="media-show__wrapper">
          <FilmDescriptionItem className="media-show__summary" text="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Architecto
            commodi
            doloribus iusto numquam temporibus veritatis, voluptatem! Atque dicta eligendi eveniet explicabo, illum
            impedit labore laborum magni porro quod sit suscipit temporibus ut!"/>

          <footer className="media-show__footer">
            <FilmDescriptionItem className="media-show__director" text="Director: ASDDDSASD"/>
            <FilmDescriptionItem className="media-show__cast" text="Cast: ljhljnljnln"/>
          </footer>

        </div>
      </div>
    );
  }
}
