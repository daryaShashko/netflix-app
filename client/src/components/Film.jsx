import React from 'react';
import PropTypes from 'prop-types';
import { FilmDescriptionItem } from './FilmDescriptionItem.jsx';
import { FilmPoster } from './FilmPoster.jsx';

export default class Film extends React.PureComponent {
  render() {
    return (
      <div className="media-show">
        <div className="media-show__wrapper">
          <FilmPoster poster={this.props.poster} show_title={this.props.showTitle}/>

          <div className="media-show__description">
            <header className="media-show__header">
              <FilmDescriptionItem className="media-show__title" text={this.props.showTitle}/>
              <FilmDescriptionItem className="media-show__rating" text={this.props.rating}/>
              <FilmDescriptionItem className="media-show__category" text={this.props.category}/>
              <FilmDescriptionItem className="media-show__release-year" text={this.props.releaseYear}/>
              <FilmDescriptionItem className="media-show__runtime" text={this.props.runtime}/>
            </header>

            <FilmDescriptionItem className="media-show__summary" text={this.props.summary}/>

            <footer className="media-show__footer">
              <FilmDescriptionItem className="media-show__director" text={`Director: ${this.props.director}`}/>
              <FilmDescriptionItem className="media-show__cast" text={`Cast: ${this.props.showCast}`}/>
            </footer>
          </div>
        </div>
      </div>
    );
  }
}

Film.defaultProps = {
  poster: '',
  showTitle: 'Button',
  rating: '3.5',
  releaseYear: '2017',
  runtime: '345',
  category: 'Horror',
  summary: 'summary',
  director: 'director,',
  showCast: 'showCast',
};

Film.propTypes = {
  poster: PropTypes.string,
  showTitle: PropTypes.string,
  rating: PropTypes.string,
  releaseYear: PropTypes.string,
  runtime: PropTypes.string,
  category: PropTypes.string,
  summary: PropTypes.string,
  director: PropTypes.string,
  showCast: PropTypes.string,
};