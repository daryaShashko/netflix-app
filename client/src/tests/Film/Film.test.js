/* eslint no-undef:0 */
import React from 'react';
import { configure , mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Film from '../../components/Film.jsx';

import Adapter from 'enzyme-adapter-react-15';

configure({ adapter: new Adapter() });

const setup = () => {
  const props = {
    movie: {
      poster: '/path/to/poster',
      showTitle: 'Title',
      rating: 4.7,
      category: 'Action',
      releaseYear: '2007-02-03',
      runtime: 96,
      summary: 'Really great film!'
    }
  };

  const enzymeWrapper = mount(
      <Film {...props} />
  );

  return {
    props,
    enzymeWrapper
  };
};

describe('Film component', () => {
  const { enzymeWrapper } = setup();

  it('should render', () => {
    expect(toJson(enzymeWrapper)).toMatchSnapshot();
  });
});