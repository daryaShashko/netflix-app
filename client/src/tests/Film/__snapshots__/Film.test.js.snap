// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film component should render 1`] = `
<Film
  category="Horror"
  director="director,"
  movie={
    Object {
      "category": "Action",
      "poster": "/path/to/poster",
      "rating": 4.7,
      "releaseYear": "2007-02-03",
      "runtime": 96,
      "showTitle": "Title",
      "summary": "Really great film!",
    }
  }
  poster=""
  rating="3.5"
  releaseYear="2017"
  runtime="345"
  showCast="showCast"
  showTitle="Button"
  summary="summary"
>
  <div
    className="media-show"
  >
    <div
      className="media-show__wrapper"
    >
      <FilmPoster
        poster=""
        showTitle="Batman"
        show_title="Button"
      >
        <div
          className="media-show__poster"
        >
          <img
            alt="Batman"
            src=""
          />
        </div>
      </FilmPoster>
      <div
        className="media-show__description"
      >
        <header
          className="media-show__header"
        >
          <FilmDescriptionItem
            className="media-show__title"
            text="Button"
          >
            <span
              className="media-show__title"
            >
              Button
            </span>
          </FilmDescriptionItem>
          <FilmDescriptionItem
            className="media-show__rating"
            text="3.5"
          >
            <span
              className="media-show__rating"
            >
              3.5
            </span>
          </FilmDescriptionItem>
          <FilmDescriptionItem
            className="media-show__category"
            text="Horror"
          >
            <span
              className="media-show__category"
            >
              Horror
            </span>
          </FilmDescriptionItem>
          <FilmDescriptionItem
            className="media-show__release-year"
            text="2017"
          >
            <span
              className="media-show__release-year"
            >
              2017
            </span>
          </FilmDescriptionItem>
          <FilmDescriptionItem
            className="media-show__runtime"
            text="345"
          >
            <span
              className="media-show__runtime"
            >
              345
            </span>
          </FilmDescriptionItem>
        </header>
        <FilmDescriptionItem
          className="media-show__summary"
          text="summary"
        >
          <span
            className="media-show__summary"
          >
            summary
          </span>
        </FilmDescriptionItem>
        <footer
          className="media-show__footer"
        >
          <FilmDescriptionItem
            className="media-show__director"
            text="Director: director,"
          >
            <span
              className="media-show__director"
            >
              Director: director,
            </span>
          </FilmDescriptionItem>
          <FilmDescriptionItem
            className="media-show__cast"
            text="Cast: showCast"
          >
            <span
              className="media-show__cast"
            >
              Cast: showCast
            </span>
          </FilmDescriptionItem>
        </footer>
      </div>
    </div>
  </div>
</Film>
`;
